portality/events/consumers/application_publisher_revision_notify.py

```diff
@@ -0,0 +1,52 @@
+from flask import url_for
+
+from portality.events.consumer import EventConsumer
+from portality import constants
+from portality import models
+from portality.lib import edges
+from portality.bll import DOAJ, exceptions
+from portality.lib.seamless import SeamlessException
+
+
+class ApplicationPublisherRevisionNotify(EventConsumer):
+    ID = "application:publisher:revision:notify"
+
+    @classmethod
+    def consumes(cls, event):
+        return event.id == constants.EVENT_APPLICATION_STATUS and \
+                event.context.get("old_status") != constants.APPLICATION_STATUS_REVISIONS_REQUIRED and \
+                event.context.get("new_status") == constants.APPLICATION_STATUS_REVISIONS_REQUIRED
```

Comment:
I would check for the presence of an `application` context key here too

Reply:
I'm checking for that in a `consume` method consistently with other notification consumers' implementation.